artichoke-backend/src/extn/core/array/backend/aggregate.rs

```diff
@@ -337,4 +339,18 @@ impl ArrayType for Aggregate {
         }
         Ok(Box::new(Self::with_parts(parts)))
     }
+
+    fn shuffle(&mut self, interp: &Artichoke) -> Result<(), Box<dyn RubyException>> {
+        let _ = interp;
+
+        let mut buffer = Vec::with_capacity(self.0.len());
+        for idx in 0..self.0.len() {
+            buffer.push(self.0.get(idx).unwrap());
+        }
+
+        let mut rng = rand::thread_rng();
+        buffer.shuffle(&mut rng);
```

Comment:
There two ways to make the result of the shuffle take effect.

The first is to call backend::buffer::Buffer::from(buffer) and replace the parts of the aggregate with this single part.

The alternative approach is to introduce a `realloc` out parameter to the shuffle signature similar to pop. The realloc parameter allows the aggregate to signal to the `Array` container that it needs to change representation. You would pass the buffer back to the container this way.